### ISSUE TITLE
3D classification tutorial Early stopping

### ISSUE THREAD

**Issue by addcue** [Project-MONAI/tutorials#666]
**Describe the bug**
The training for the tutorial 3D classification terminates after 5 epochs. Pytorch ignite EarlyStopping handler is used to stop training. This occurs for both the files for training_dict and training_array. 

<details> <summary> logs </summary> 
./3d_classification/ignite/densenet_training_dict.py
MONAI version: 0.8.1
Numpy version: 1.22.3
Pytorch version: 1.11.0+cu102
MONAI flags: HAS_EXT = False, USE_COMPILED = False
MONAI rev id: 71ff399a3ea07aef667b23653620a290364095b1

Optional dependencies:
Pytorch Ignite version: 0.4.8
Nibabel version: 3.2.2
scikit-image version: 0.19.2
Pillow version: 9.1.0
Tensorboard version: 2.8.0
gdown version: 4.4.0
TorchVision version: 0.12.0+cu102
tqdm version: 4.64.0
lmdb version: 1.3.0
psutil version: 5.9.0
pandas version: 1.4.2
einops version: 0.4.1
transformers version: 4.18.0
mlflow version: 1.25.1

For details about installing the optional dependencies, please visit:
    https://docs.monai.io/en/latest/installation.html#installing-the-recommended-dependencies

torch.Size([2, 1, 96, 96, 96]) tensor([0, 0])
INFO:ignite.engine.engine.Engine:Engine run starting with max_epochs=30.
INFO:trainer:Epoch: 1/30, Iter: 1/5 -- Loss: 0.8398 
INFO:trainer:Epoch: 1/30, Iter: 2/5 -- Loss: 0.7421 
INFO:trainer:Epoch: 1/30, Iter: 3/5 -- Loss: 0.6903 
INFO:trainer:Epoch: 1/30, Iter: 4/5 -- Loss: 0.6971 
INFO:trainer:Epoch: 1/30, Iter: 5/5 -- Loss: 0.8217 
INFO:ignite.engine.engine.Engine:Engine run starting with max_epochs=1.
INFO:evaluator:Epoch[1] Metrics -- AUC: 0.7500 
INFO:ignite.engine.engine.Engine:Epoch[1] Complete. Time taken: 00:00:05
INFO:ignite.engine.engine.Engine:Engine run complete. Time taken: 00:00:05
INFO:ignite.engine.engine.Engine:Epoch[1] Complete. Time taken: 00:00:23
INFO:trainer:Epoch: 2/30, Iter: 1/5 -- Loss: 0.7278 
INFO:trainer:Epoch: 2/30, Iter: 2/5 -- Loss: 0.8066 
INFO:trainer:Epoch: 2/30, Iter: 3/5 -- Loss: 0.7281 
INFO:trainer:Epoch: 2/30, Iter: 4/5 -- Loss: 0.7249 
INFO:trainer:Epoch: 2/30, Iter: 5/5 -- Loss: 0.6370 
INFO:ignite.engine.engine.Engine:Engine run starting with max_epochs=1.
INFO:evaluator:Epoch[2] Metrics -- AUC: 0.7500 
INFO:ignite.engine.engine.Engine:Epoch[1] Complete. Time taken: 00:00:05
INFO:ignite.engine.engine.Engine:Engine run complete. Time taken: 00:00:06
INFO:ignite.engine.engine.Engine:Epoch[2] Complete. Time taken: 00:00:22
INFO:trainer:Epoch: 3/30, Iter: 1/5 -- Loss: 0.7986 
INFO:trainer:Epoch: 3/30, Iter: 2/5 -- Loss: 0.7600 
INFO:trainer:Epoch: 3/30, Iter: 3/5 -- Loss: 0.5949 
INFO:trainer:Epoch: 3/30, Iter: 4/5 -- Loss: 0.7725 
INFO:trainer:Epoch: 3/30, Iter: 5/5 -- Loss: 0.6432 
INFO:ignite.engine.engine.Engine:Engine run starting with max_epochs=1.
INFO:evaluator:Epoch[3] Metrics -- AUC: 0.7083 
INFO:ignite.engine.engine.Engine:Epoch[1] Complete. Time taken: 00:00:05
INFO:ignite.engine.engine.Engine:Engine run complete. Time taken: 00:00:06
INFO:ignite.engine.engine.Engine:Epoch[3] Complete. Time taken: 00:00:23
INFO:trainer:Epoch: 4/30, Iter: 1/5 -- Loss: 0.7858 
INFO:trainer:Epoch: 4/30, Iter: 2/5 -- Loss: 0.7818 
INFO:trainer:Epoch: 4/30, Iter: 3/5 -- Loss: 0.6025 
INFO:trainer:Epoch: 4/30, Iter: 4/5 -- Loss: 0.5577 
INFO:trainer:Epoch: 4/30, Iter: 5/5 -- Loss: 0.7618 
INFO:ignite.engine.engine.Engine:Engine run starting with max_epochs=1.
INFO:evaluator:Epoch[4] Metrics -- AUC: 0.6667 
INFO:ignite.engine.engine.Engine:Epoch[1] Complete. Time taken: 00:00:06
INFO:ignite.engine.engine.Engine:Engine run complete. Time taken: 00:00:06
INFO:ignite.engine.engine.Engine:Epoch[4] Complete. Time taken: 00:00:23
INFO:trainer:Epoch: 5/30, Iter: 1/5 -- Loss: 0.7667 
INFO:trainer:Epoch: 5/30, Iter: 2/5 -- Loss: 0.7293 
INFO:trainer:Epoch: 5/30, Iter: 3/5 -- Loss: 0.5294 
INFO:trainer:Epoch: 5/30, Iter: 4/5 -- Loss: 0.7414 
INFO:trainer:Epoch: 5/30, Iter: 5/5 -- Loss: 0.5467 
INFO:ignite.engine.engine.Engine:Engine run starting with max_epochs=1.
INFO:evaluator:Epoch[5] Metrics -- AUC: 0.7500 
2022-04-21 00:17:29,358 ignite.handlers.early_stopping.EarlyStopping INFO: EarlyStopping: Stop training
INFO:ignite.engine.engine.Engine:Terminate signaled. Engine will stop after current iteration is finished.
INFO:ignite.engine.engine.Engine:Epoch[1] Complete. Time taken: 00:00:05
INFO:ignite.engine.engine.Engine:Engine run complete. Time taken: 00:00:06
INFO:ignite.engine.engine.Engine:Epoch[5] Complete. Time taken: 00:00:22
INFO:ignite.engine.engine.Engine:Engine run complete. Time taken: 00:01:53
State:
	iteration: 25
	epoch: 5
	epoch_length: 5
	max_epochs: 30
	output: 0.5467474460601807
	batch: <class 'dict'>
	metrics: <class 'dict'>
	dataloader: <class 'torch.utils.data.dataloader.DataLoader'>
	seed: <class 'NoneType'>
	times: <class 'dict'>

examples 3d classification ignite: model file not generated
Running ./3d_classification/ignite/densenet_evaluation_dict.py
MONAI version: 0.8.1
Numpy version: 1.22.3
Pytorch version: 1.11.0+cu102
MONAI flags: HAS_EXT = False, USE_COMPILED = False
MONAI rev id: 71ff399a3ea07aef667b23653620a290364095b1

Optional dependencies:
Pytorch Ignite version: 0.4.8
Nibabel version: 3.2.2
scikit-image version: 0.19.2
Pillow version: 9.1.0
Tensorboard version: 2.8.0
gdown version: 4.4.0
TorchVision version: 0.12.0+cu102
tqdm version: 4.64.0
lmdb version: 1.3.0
psutil version: 5.9.0
pandas version: 1.4.2
einops version: 0.4.1
transformers version: 4.18.0
mlflow version: 1.25.1

For details about installing the optional dependencies, please visit:
    https://docs.monai.io/en/latest/installation.html#installing-the-recommended-dependencies

INFO:ignite.engine.engine.Engine:Engine run starting with max_epochs=1.
INFO:ignite.engine.engine.Engine:Restored all variables from ./runs_dict/net_checkpoint_20.pt
INFO:evaluator:Epoch[1] Metrics -- Accuracy: 0.4000 
INFO:ignite.engine.engine.Engine:Epoch[1] Complete. Time taken: 00:00:05
INFO:ignite.engine.engine.Engine:Engine run complete. Time taken: 00:00:06
State:
	iteration: 5
	epoch: 1
	epoch_length: 5
	max_epochs: 1
	output: <class 'tuple'>
	batch: <class 'dict'>
	metrics: <class 'dict'>
	dataloader: <class 'torch.utils.data.dataloader.DataLoader'>
	seed: <class 'NoneType'>
	times: <class 'dict'>
</details>

**Environment (please complete the following information):**
 - Python version: 3.8.10
 - MONAI version:0.8.1

**Comment by Nic-Ma** [comment]
Hi @addcue ,

I think this is expected behavior to show the usage of `EarlyStopping` handler.
Or maybe I misunderstood your question?

Thanks.

**Comment by addcue** [comment]
Hi,

is there a tutorial for using the classification in its entirely without the early stopping?

**Comment by Nic-Ma** [comment]
Hi @addcue ,

If you don't want to apply the Early-Stop, maybe you can remove these 2 lines and run again:
https://github.com/Project-MONAI/tutorials/blob/master/3d_classification/ignite/densenet_training_dict.py#L151-L152

Thanks.

**Comment by addcue** [comment]
Thanks! That helped fix it for the training_dict file, but in the training_array walkthrough the model is not improving. The AUC is stuck at 0.6. Do you know why that’s happening?

From: Nic Ma ***@***.***>
Reply-To: Project-MONAI/tutorials ***@***.***>
Date: Thursday, April 21, 2022 at 9:48 AM
To: Project-MONAI/tutorials ***@***.***>
Cc: "Qiu, Edwin" ***@***.***>, Mention ***@***.***>
Subject: Re: [Project-MONAI/tutorials] 3D classification tutorial Early stopping (Issue #666)


Hi @addcue<https://github.com/addcue> ,

If you don't want to apply the Early-Stop, maybe you can remove these 2 lines and run again:
https://github.com/Project-MONAI/tutorials/blob/master/3d_classification/ignite/densenet_training_dict.py#L151-L152

Thanks.

**Comment by Nic-Ma** [comment]
Hi @addcue ,

These examples only try to show the functionality, not real-world training, you can see that we only used few images of the dataset.

Thanks.
